ee/experimental/incident_utils.py

```diff
@@ -1,10 +1,152 @@
+import os
+import logging
+
 import numpy as np
 import pandas as pd
 import networkx as nx
 
 from typing import List
+from openai import OpenAI
+
+from datetime import datetime, timedelta
+
+from fastapi import Depends
+
+from ee.experimental.note_utils import NodeCandidateQueue, NodeCandidate
+from ee.experimental.graph_utils import create_graph
+from ee.experimental.statistical_utils import get_alert_pmi_matrix
+
+from pusher import Pusher
+
+from keep.api.models.db.alert import Alert, Incident
+from keep.api.core.db import (
+    assign_alert_to_incident,
+    is_alert_assigned_to_incident,
+    get_last_alerts,
+    create_incident_from_dict,
+    update_incident_summary,
+)
+
+from keep.api.core.dependencies import (
+    AuthenticatedEntity,
+    AuthVerifier,
+    get_pusher_client,
+)
+
+from keep.api.core.db import (
+    assign_alert_to_incident,
+    create_incident_from_dict,
+    get_incident_by_id,
+    get_last_alerts,
+    get_last_incidents,
+    write_pmi_matrix_to_db,
+)
+
+logger = logging.getLogger(__name__)
+
+ALGORITHM_VERBOSE_NAME = "Basic correlation alrithm v0.2"
+
+
+def calculate_pmi_matrix(
+    ctx: dict | None,  # arq context
+    tenant_id: str,
+    upper_timestamp: datetime = datetime.now() - timedelta(seconds=60 * 60),
```

Comment:
@VladimirFilonov commented about this a while back, this is not a good practice, see:
![CleanShot 2024-08-11 at 18 16 50@2x](https://github.com/user-attachments/assets/b7ab8f56-36e2-4990-8e47-0ed463e3e02e)